### PR TITLE
Change nextHopProtocol to a ByteString

### PR DESCRIPTION
**Pull request by marcoscaceres**
closes #277 

Recreated #277 because of all the conflicts.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/resource-timing/pull/281.html" title="Last updated on Jul 7, 2021, 1:15 AM UTC (e71ab98)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/resource-timing/281/edea639...e71ab98.html" title="Last updated on Jul 7, 2021, 1:15 AM UTC (e71ab98)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -592,11 +592,11 @@ <h3>
...
         <p data-dfn-for="PerformanceResourceTiming">
           The <dfn>nextHopProtocol</dfn> getter steps are to [=/isomorphic
-          decode=] <a>this</a>'s <a data-for="PerformanceResourceTiming">timing
-          info</a>'s [=fetch timing info/final connection timing info=]'s
-          [=connection timing info/ALPN negotiated protocol=]. See
-          <a data-cite="FETCH#record-connection-timing-info">Recording
-          connection timing info</a> for more info.
+          decode=] <a>this</a>'s [=PerformanceResourceTiming/timing info=]'s
+          [=fetch timing info/final connection timing info=]'s [=connection
+          timing info/ALPN negotiated protocol=]. See <a data-cite=
```
Comment (yoavweiss) [inline index.html:597]: Can we also drop the data-cite here?

**Review by yoavweiss — APPROVED** [review]
LGTM

File: index.html (hunk 1 above)
Comment (marcoscaceres) [reply index.html:597]: Possibly, but it's explicitly marked as a "private" definition (noexport) by Fetch. @annevk, not sure if you want to export that? Do you have a suggestion of what we could link to?

Comment (yoavweiss) [reply index.html:597]: OK, let's not block on it then. We can open a separate issue for this

**Review by yoavweiss — APPROVED** [review]
LGTM

**Pull request merged**